### PR TITLE
[Resources and support] Revise how the search works

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR modifies the way the search on `/resources/search/` works. The changes include -

- Only use account article titles as the search criteria
- Split the article titles into a list of keywords, so that we can check to see if any keyword "startsWith" a keyword from the user query
- Un-pluralize queries ending with "s" or "ies"

https://github.com/department-of-veterans-affairs/va.gov-team/issues/15650
https://github.com/department-of-veterans-affairs/va.gov-team/issues/15598

## Testing done
Did a bunch of queries and confirmed the results were as expected

## Screenshots
image

## Acceptance criteria
- [ ] Search results are improved

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/resources-and-support/hooks/useArticleData.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import { useEffect, useState } from 'react';
 import * as Sentry from '@sentry/browser';
 
+// We use CMS demo environments to publish articles in bulk.
+// This means that at this time, we can't use the environment.BASE_URL,
+// because the CMS demo env uses domains not defined there, and we need
+// this search app to align with the demo.
+const baseUrl = document.location.origin;
```
Comment (kelsonic) [inline src/applications/resources-and-support/hooks/useArticleData.js:8]: Thanks for this comment! 🔥

File: src/applications/resources-and-support/hooks/useGetSearchResults.js [hunk 2]
```diff
@@ -15,7 +15,11 @@ export default function useGetSearchResults(articles, query, page) {
         .split(' ')
         .map(keyword => keyword.toLowerCase())
         .map(keyword => {
-          if (keyword.length > 3 && keyword.endsWith('s')) {
+          if (keyword.length > 6 && keyword.endsWith('ies')) {
```
Comment (kelsonic) [inline src/applications/resources-and-support/hooks/useGetSearchResults.js:18]: How come we bumped this from 3 to 6 out of curiosity? 🙂

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: src/applications/resources-and-support/hooks/useGetSearchResults.js (hunk 2 above)
Comment (ncksllvn) [reply src/applications/resources-and-support/hooks/useGetSearchResults.js:18]: I think the diff is a little misleading. It's a char length of 6 if we are stripping off an entire `ies`, but char length of 3 if just an `s`. The 6 doesn't really have any special meaning. I picked it just bc it seemed reasonable for a word ending with `ies`

File: src/applications/resources-and-support/hooks/useArticleData.js (hunk 1 above)
Comment (ncksllvn) [reply src/applications/resources-and-support/hooks/useArticleData.js:8]: It's still going to confuse me in a few weeks from now lol

**Pull request merged**